packages/core/src/use-cases/vault-lifecycle/lock-vault.ts

```
@@ -75,8 +77,20 @@ export class LockVaultUseCase {
       if (cleanupError !== undefined) {
         throw cleanupError;
       }
+    } catch (error) {
+      executionError = error;
     } finally {
-      await this.unlockedVaultRepository.removeUnlockedVault();
+      try {
+        await this.removeUnlockedVaultSession.execute();
+      } catch (error) {
+        if (executionError === undefined) {
+          throw error;
```

Biome (2.4.16)

[error] Unsafe usage of 'throw'.

(lint/correctness/noUnsafeFinally)